File: EmployeePayrollWeb/src/app/classes/advance.ts
export interface IAdvance{
    empName: string;
    empContactNo: string;
    employeeId: number;
    advanceId: number;
    advanceDate: string;
    advanceAmount: number;
    reason: string;
}
export class advance {
    advanceId: number = 0;
    employeeId: number = 0;
    advanceDate?: Date = undefined;
    advanceAmount: number = 0;
    reason: string = "";
  }

File: EmployeePayroll.API/Models/Advance.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EmployeePayroll.API.Models
{
    [Table("Advance")]
    public class Advance
    {
        [Key]
        public int AdvanceId { get; set; }
        public int? EmployeeId { get; set; }
        public DateTime? AdvanceDate { get; set; }
        [Precision(18, 2)]
        public decimal AdvanceAmount { get; set; } = 0;
        public string? Reason { get; set; }
    }
}
